File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        long ans = 0;
        while(x){
            int last = x%10;
            x /= 10;
            ans = ans*10 + last;
            if(ans > INT_MAX || ans < INT_MIN) return 0;
        }   
        return (int) ans;
    }
};

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        sign = (x > 0) - (x < 0)
        
        reversed_x = sign * int(str(abs(x))[::-1])

        if reversed_x < -2**31 or reversed_x > 2**31 - 1:
            return 0
            
        return reversed_x